flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -435,9 +436,13 @@ public void killCluster(ApplicationId applicationId) throws FlinkException {
 			UserGroupInformation loginUser = UserGroupInformation.getCurrentUser();
 			if (loginUser.getAuthenticationMethod() == UserGroupInformation.AuthenticationMethod.KERBEROS
 					&& useTicketCache && !loginUser.hasKerberosCredentials()) {
-				LOG.error("Hadoop security with Kerberos is enabled but the login user does not have Kerberos credentials");
-				throw new RuntimeException("Hadoop security with Kerberos is enabled but the login user " +
+				// a delegation token is an adequate substitute in most cases
```

Comment:
You added a stricter condition to the exception identification. Comparing with the origin logic, if it does not meet the conditions you added, the original exception will not be logged and thrown. Is this what you expected? In addition, if you just want to identify the exception more strictly, why not add it to the outer if block together?

Reply:
Yes. this is the expected result. See: https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/HadoopModule.java#L146 which is used in client/JM/TM. 

However I think the checker in here not present - this means specifically in YarnClusterDescriptor it does not support delegation token. This seems like a bug to me. 

Regarding the implementation, yes I can do that as well (e.g. merging with the upper layer if, I was trying to pair the code logic with `HadoopModule`.

Reply:
We could also move the whole block, i.e. 
```
if (securityConfig.useTicketCache() && !loginUser.hasKerberosCredentials()) {
					// a delegation token is an adequate substitute in most cases
					if (!HadoopUtils.hasHDFSDelegationToken()) {
						LOG.warn("Hadoop security is enabled but current login user does not have Kerberos credentials");
					}
				}
```
to `HadoopUtils` and use it in both `HadoopModule` and this code, thus reducing code duplication.

Reply:
+1 moving to HadoopUtils makes more sense to me actually - if I understand correctly, delegation Token is a hadoop concept; Kerberos official doc seems to only contain a guidance on credential delegation: https://tools.ietf.org/html/rfc5896.html.

I will make the adjustment accordingly